internal/service/opensearchserverless/security_policy_data_source.go

```diff
@@ -0,0 +1,72 @@
+package opensearchserverless
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+)
+
+// @SDKDataSource("aws_opensearchserverless_security_policy")
+func DataSourceSecurityPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceSecurityPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"policy_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func dataSourceSecurityPolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).OpenSearchServerlessClient(ctx)
+
+	securityPolicyName := d.Get("name").(string)
+	securityPolicyType := d.Get("type").(string)
+	securityPolicy, err := FindSecurityPolicyByNameAndType(ctx, conn, securityPolicyName, securityPolicyType)
+
+	if err != nil {
+		return sdkdiag.AppendErrorf(diags, "reading SecurityPolicy with name (%s) and type (%s): %s", securityPolicyName, securityPolicyType, err)
+	}
+
+	policyBytes, err := securityPolicy.Policy.MarshalSmithyDocument()
```

Comment:
@jar-b have you seen this method before? I have not seen Smith document marshaling prior, to want to verify.

Reply:
Yes, the OpenSearch Serverless Access Policy was just merged today and uses a similar pattern:
https://github.com/hashicorp/terraform-provider-aws/blob/2402e640fbe617bd5cc97003f09c27e38af197f2/internal/service/opensearchserverless/access_policy_data_source.go#L88